Sources/MapboxMaps/Foundation/Interpolator.swift

```diff
@@ -0,0 +1,39 @@
+internal protocol InterpolatorProtocol {
+    func interpolate(from: Double, to: Double, percent: Double) -> Double
```

Comment:
Re: using associated types: I did that initially, but it actually makes the protocol much less useful, IMO, because I can't inject an object of type InterpolatorProtocol anymore if I do that — the consuming class has to become generic.

Reply:
> Maybe progress, fraction/fractionComplete could be considered here, as percent can be interpreted as value per 100, i.e. 50, not 0.5.

I'm fine with fraction.

Reply:
Re: using associated types: Good point, in swift protocols with associated types can, indeed, be quite notorious to work with. If not taken under control generics could spread like a plague.